### PR TITLE
[NCL-4974] Fixed rsql mapping typo for BCs

### PR DESCRIPTION
**Pull request by michalovjan**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/BuildConfigurationRSQLMapper.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public BuildConfigurationRSQLMapper() {
     protected SingularAttribute<BuildConfiguration, ? extends GenericEntity<Integer>> toEntity(String name) {
         switch (name) {
             case "project": return BuildConfiguration_.project;
-            case "repositoryConfiguration": return BuildConfiguration_.repositoryConfiguration;
+            case "repository": return BuildConfiguration_.repositoryConfiguration;
```
Comment (matedo1) [inline facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/BuildConfigurationRSQLMapper.java]: scmRepository should be used

Comment (janinko) [reply facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/BuildConfigurationRSQLMapper.java]: the DTO field name is `repository`

**Comment by matedo1** [comment]
retest this please

**Comment by matedo1** [comment]
retest this please

**Comment by matedo1** [comment]
retest this please

**Pull request merged**